internal/bminventory/inventory.go

```diff
@@ -1653,6 +1660,23 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func updateClusterNonePlatformParams(params installer.V2UpdateClusterParams, cluster *common.Cluster) {
+	if params.ClusterUpdateParams.UserManagedNetworking == nil || cluster.Platform == nil || cluster.UserManagedNetworking == nil {
+		return
+	}
+
+	if *params.ClusterUpdateParams.UserManagedNetworking && !*cluster.UserManagedNetworking {
+		if *cluster.Platform.Type == models.PlatformTypeBaremetal {
+			params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
+
+		} else if params.ClusterUpdateParams.Platform != nil && models.PlatformTypeBaremetal == *params.ClusterUpdateParams.Platform.Type {
+			params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
+		}
+	} else if params.ClusterUpdateParams.Platform != nil && *params.ClusterUpdateParams.Platform.Type == models.PlatformTypeNone {
+		params.ClusterUpdateParams.UserManagedNetworking = swag.Bool(true)
+	}
```

Comment:
What about the case where user sends `UserManagedNetworking = false`, and `cluster.UserManagedNetworking = true`?
it seems that that case is ignored.
Also, if we are not to returning badRequest we should make it clear what param takes presedence Platform or UMN flag.

Reply:
> What about the case where user sends UserManagedNetworking = false, and cluster.UserManagedNetworking = true?

You right, but I think we can handle this case instead `*params.ClusterUpdateParams.UserManagedNetworking == false && *cluster.Platform.Type == models.PlatformTypeNone`
WDYT?

Reply:
Unsure how we can handle it though.

Reply:
I think that in that case we should return to baremetal. The "duplication" of UMN and platform=none casing some edge cases that not that trivial to handle (e.g. selected vsphere platfrom with UMN=true and change back to baremetal, is it disabling the UMN?)

Reply:
These 2 parameters are optional, if the user chooses to provide both with conflicting values we shouldn't overthink it and simply return bad request.
In case one was provided we should align the other param as well (within the boundary of None / Baremetal platform and not overriding any other platforms provided by the user).